internal/resources/diffValidation/resource_nsxt_lb_monitor.go

```diff
@@ -0,0 +1,148 @@
+//  (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package diffvalidation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	serviceTypes   = "type"
+	httpMonitor    = "http_monitor"
+	httpsMonitor   = "https_monitor"
+	icmpMonitor    = "icmp_monitor"
+	passiveMonitor = "passive_monitor"
+	tcpMonitor     = "tcp_monitor"
+	udpMonitor     = "udp_monitor"
+)
+
+type LoadBalancerMonitor struct {
+	diff *schema.ResourceDiff
+}
+
+func NewLoadBalancerMonitorValidate(diff *schema.ResourceDiff) *LoadBalancerMonitor {
+	return &LoadBalancerMonitor{
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerMonitor) DiffValidate() error {
+	err := l.validateHttp()
+	if err != nil {
+		return err
+	}
+	err = l.validateHttps()
+	if err != nil {
+		return err
+	}
+	err = l.validateIcmp()
+	if err != nil {
+		return err
+	}
+	err = l.validatePassive()
+	if err != nil {
+		return err
+	}
+	err = l.validateTcp()
+	if err != nil {
+		return err
+	}
+	err = l.validateUdp()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *LoadBalancerMonitor) validateHttp() error {
+	serviceType := serviceTypes
+	http_monitor := httpMonitor
```

Comment:
are these lines really needed ? Can u not directly use the const's ?

Reply:
updated